force-app/main/releaseGating/classes/ReleaseGateAPIService.cls

```diff
@@ -0,0 +1,230 @@
+     * @return   A new Callable instance
+     */
+    private Callable createReleaseGateCallableInstance() {
+        String namespace = this.releaseGateRegistry.namespace == null ? '' : this.releaseGateRegistry.namespace;
+        return (Callable) Type.forName(namespace, this.releaseGateRegistry.classname).newInstance();
+    }
+
+    /**
+     * @description Retrieve a GetReleaseGateActionResultModel representing the release gates
+     * @return           A GetReleaseGateActionResultModel
+     */
+    public virtual GetReleaseGateActionResultModel getReleaseGates() {
+        Object result = this.makeCall(
+            GET_ACTION,
+            new Map<String, Object>{ VERSION_PARAMETER => this.releaseGateRegistry.apiVersion }
+        );
+        GetReleaseGateActionResultModel resultModel = (GetReleaseGateActionResultModel) JSON.deserialize(
+            (String) result,
+            GetReleaseGateActionResultModel.class
+        );
+        this.verifyCallableResult(resultModel, GET_ACTION);
+        return resultModel;
+    }
+
+    /**
+     * @description Calls the specified callable action
+     * @param  action     action name to call
+     * @param  parameters action parameters
+     * @return            Callable return result
+     */
+    protected virtual Object makeCall(String action, Map<String, Object> parameters) {
+        Callable callableInstance = this.locateReleaseGateCallable();
+        UTIL_Debug.debug(
+            String.format(
+                'Calling {0}.{1} action, version {2}',
+                new List<Object>{ this.releaseGateRegistry.name, action, parameters.get(VERSION_PARAMETER) }
+            )
+        );
+        Object result;
+        try {
+            result = callableInstance.call(action, parameters);
+        } catch (Exception ex) {
+            String errorMessage = String.format(
+                'An unhandled exception occurred during execution of {0}.{1}: {2}',
+                new List<Object>{ this.releaseGateRegistry.name, action, ex.getMessage() }
+            );
+            UTIL_Debug.debug(System.LoggingLevel.ERROR, errorMessage);
+            throw new ReleaseGateCallableException(
+                'An unhandled exception occurred during execution: ' + errorMessage,
+                ex,
+                this.releaseGateRegistry,
+                null
+            );
+        }
+        UTIL_Debug.debug(
+            String.format(
+                'Finished {0}.{1}, got the following response: \r\n {2}',
+                new List<Object>{ this.releaseGateRegistry.name, action, result }
+            )
+        );
+        return result;
+    }
+
+    /**
+     * @description Verify if the result is successful, throw ReleaseGateCallableException if not
+     * @param  resultModel Callable action result
+     * @param  action      Callable action name
+     */
+    private void verifyCallableResult(CallableResult resultModel, String action) {
+        if (!resultModel.success) {
```

Comment:
Please change this to a guard clause so we can force code coverage. Consider changing the method to a Boolean return to allow for easier testing.
```suggestion
        if (resultModel.success) {
            return;
        }
```

Reply:
Done